frontend/src/App.js

```diff
@@ -2,9 +2,45 @@ import React, { Component } from 'react';
 import './App.css';
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      quote: 'They don\'t know that we know they know we know.',
+      typedText: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    const typedText = event.target.value;
+    this.setState({ typedText: typedText });
+  }
+
+  getDisplayQuote() {
+    const quoteArray = this.state.quote.split('');
+    const typedArray = this.state.typedText.split('');
+    return (
+      quoteArray.map((letter, index) => {
+        if (!(index in typedArray)) {
+          return letter;
+        }
+        if (letter === typedArray[index]) {
+          return <span className="right-letter">{letter}</span>;
+        }
+        return <span className="wrong-letter">{letter}</span>;
+      })
+    );
+  }
```

Comment:
Structurally, this function might be doing too much. It has 2 jobs right now

- it is calculating the correctness of the input string
- it is rendering the appropriate JSX

For readability, I recommend you put the rendering logic inside of the `render` method and have this function return a boolean that indicates the validity of the input.

This keeps the rendering logic in a single place and makes this function more modular